### PR TITLE
nrn_setup: some cleaning

### PR DESCRIPTION
**Pull request by alkino**

### CODE REVIEW

File: coreneuron/nrniv/nrn_setup.cpp [hunk 1]
```diff
@@ -1720,21 +1720,19 @@ for (int i=0; i < nt.end; ++i) {
...
     std::vector<BAMech*> bamap(memb_func.size());
     for (int i = 0; i < BEFORE_AFTER_SIZE; ++i) {
-        BAMech* bam;
-        NrnThreadBAList *tbl, **ptbl;
         for (int ii = 0; ii < memb_func.size(); ++ii) {
-            bamap[ii] = (BAMech*)0;
+            bamap[ii] = nullptr;
         }
-        for (bam = corenrn.get_bamech()[i]; bam; bam = bam->next) {
+        for (BAMech* bam = corenrn.get_bamech()[i]; bam; bam = bam->next) {
```
Comment (ohm314) [inline coreneuron/nrniv/nrn_setup.cpp]:
```suggestion
        for (auto bam = corenrn.get_bamech()[i]; bam; bam = bam->next) {
```
that works too, no?

**Comment by alkino** [comment]
please retest

**Comment by alkino** [comment]
Please retest

**Comment by alkino** [comment]
Please retest

**Pull request merged**